MaxText/layers/quantizations.py

```diff
@@ -202,6 +205,84 @@ def dot_general_cls(self, mesh_axes: Tuple[str, ...] = ()):
     return nn.Fp8DotGeneralOp
 
 
+class Fp8Einsum(nn.Module):
```

Comment:
Nit: we already have the Quantization superclass above. What if we added a `einsum` method to that and then added this logic as part of the `einsum` implementation within `Fp8Quantization`. We could avoid some of that branching we need in `linears` up above that way?

Reply:
Emm, I tried to move this logic to Fp8Quantization. But the trick part is that we need to maintain some variables. So, I use this inherited nn.Module. When moving the logic to Fp8Quantization and I always got errors like `TypeError: Can't call __hash__ on modules that hold variables.`.

Reply:
Yes, this can be done. PTAL.